android/build.gradle

```diff
@@ -81,69 +77,83 @@ task sourcesJar(type: Jar) {
+            from components.android
+
+            artifact javadocJar
+            artifact sourcesJar
+
+            pom {
+                name = project.group + ":" + project.name
+                url = 'https://github.com/grpc/grpc-java'
+                afterEvaluate {
+                    // description is not available until evaluated.
+                    description = project.description
+                }
+
+                scm {
+                    connection = 'scm:git:https://github.com/grpc/grpc-java.git'
+                    developerConnection = 'scm:git:git@github.com:grpc/grpc-java.git'
+                    url = 'https://github.com/grpc/grpc-java'
+                }
+
+                licenses {
+                    license {
+                        name = 'Apache 2.0'
+                        url = 'https://opensource.org/licenses/Apache-2.0'
+                    }
+                }
+
+                developers {
+                    developer {
+                        id = "grpc.io"
+                        name = "gRPC Contributors"
+                        email = "grpc-io@googlegroups.com"
+                        url = "https://grpc.io/"
+                        organization = "gRPC Authors"
+                        organizationUrl = "https://www.google.com"
+                    }
+                }
+
+                withXml {
+                    asNode().dependencies.'*'.findAll() { dep ->
+                        dep.artifactId.text() in ['grpc-api', 'grpc-core']
+                    }.each() { core ->
+                        core.version*.value = "[" + core.version.text() + "]"
+                    }
+                }
             }
         }
-        repository(url: stagingUrl, configureAuth)
-        snapshotRepository(url: 'https://oss.sonatype.org/content/repositories/snapshots/', configureAuth)
     }
-}
-
-[
-    install.repositories.mavenInstaller,
-    uploadArchives.repositories.mavenDeployer,
-]*.pom*.whenConfigured { pom ->
-    pom.project {
-        name "$project.group:$project.name"
-        description project.description
-        url 'https://conscrypt.org/'
-
-        scm {
-            connection 'scm:git:https://github.com/grpc/grpc-java.git'
-            developerConnection 'scm:git:git@github.com:grpc/grpc-java.git'
-            url 'https://github.com/grpc/grpc-java'
-        }
-
-        licenses {
-            license {
-                name 'Apache 2.0'
-                url 'https://opensource.org/licenses/Apache-2.0'
-            }
-        }
-
-        developers {
-            developer {
-                id "grpc.io"
-                name "gRPC Contributors"
-                email "grpc-io@googlegroups.com"
-                url "https://grpc.io/"
-                organization = "gRPC Authors"
-                organizationUrl "https://www.google.com"
+    repositories {
+        maven {
+            if (rootProject.hasProperty('repositoryDir')) {
```

Comment:
where is `repositoryDir` set? `android.sh`?

Reply:
It is within linux_artifacts.sh. It's more obvious now because I needed to change the line that actually includes it.